lib/cloud/azure/vm.go

```diff
@@ -166,3 +167,74 @@ func (c *vmClient) ListVirtualMachines(ctx context.Context, resourceGroup string
 
 	return virtualMachines, nil
 }
+
+// vmRunCommand provides an interface for an Azure Run Command client.
+type vmRunCommand interface {
+	BeginCreateOrUpdate(ctx context.Context, resourceGroup, vmName, runCommandName string, runCommand armcompute.VirtualMachineRunCommand, options *armcompute.VirtualMachineRunCommandsClientBeginCreateOrUpdateOptions) (*runtime.Poller[armcompute.VirtualMachineRunCommandsClientCreateOrUpdateResponse], error)
+}
+
+// RunCommandRequest combines parameters for running a command on an Azure virtual machine.
+type RunCommandRequest struct {
+	// Region is the region of the VM.
+	Region string
+	// ResourceGroup is the resource group for the VM.
+	ResourceGroup string
+	// VMName is the name of the VM.
+	VMName string
+	// Script is the URI of the script for the virtual machine to execute.
+	Script string
+	// Parameters is a list of parameters for the script.
+	Parameters []string
+}
+
+// RunCommandClient is a client for Azure Run Commands.
+type RunCommandClient interface {
+	Run(ctx context.Context, req RunCommandRequest) error
+}
+
+type runCommandClient struct {
+	api vmRunCommand
+}
+
+// NewRunCommandClient creates a new Azure Run Command client by subscription
+// and credentials.
+func NewRunCommandClient(subscription string, cred azcore.TokenCredential, options *arm.ClientOptions) (RunCommandClient, error) {
+	runCommandAPI, err := armcompute.NewVirtualMachineRunCommandsClient(subscription, cred, options)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+	return NewRunCommandClientByAPI(runCommandAPI), nil
+}
+
+// NewRunCommandClientByAPI creates a new Azure Run Command client by ARM API
+// client.
+func NewRunCommandClientByAPI(api vmRunCommand) RunCommandClient {
+	return &runCommandClient{
+		api: api,
+	}
+}
+
+// Run runs a command on a virtual machine.
+func (c *runCommandClient) Run(ctx context.Context, req RunCommandRequest) error {
```

Comment:
Is there a reasonable way to unit test this?

Reply:
I don't think so. `Run` is just a wrapper the Azure SDK.